pkg/operator/ceph/nfs/config.go

```diff
@@ -126,3 +133,231 @@ RADOS_URLS {
+		return errors.Wrapf(err, "failed write ganesha kerberos configuration block temp file for %s", radosInfoStr)
+	}
+
+	radosFlags := []string{
+		"--pool", radosPool,
+		"--namespace", radosNs,
+	}
+
+	// write ganesha kerberos configuration block to rados object from temp file
+	cmd := cephclient.NewRadosCommand(context, clusterInfo,
+		append(radosFlags, "put", kerberosRadosObjectName, krbBlockFile.Name()))
+	_, err = cmd.RunWithTimeout(exec.CephCommandsTimeout)
+	if err != nil {
+		return errors.Wrapf(err, "failed to create or update the ganesha kerberos config object %s/%s",
+			radosInfoStr, kerberosRadosObjectName)
+	}
+
+	// prepend the config block that includes the kerberos config object to the ganesha config object
+	ganeshaConfigObjName := getGaneshaConfigObject(nfs)
+	krbIncludeBlock := ganeshaConfigIncludeKrbBlock(nfs, kerberosRadosObjectName)
+	err = atomicPrependToConfigObject(context, clusterInfo, radosPool, radosNs, ganeshaConfigObjName, krbIncludeBlock)
+	if err != nil {
+		return errors.Wrapf(err, "failed to update the ganesha config object to include the kerberos object config")
+	}
+
+	return nil
+}
+
+func removeKerberosRadosConfig(context *clusterd.Context, clusterInfo *cephclient.ClusterInfo, nfs *cephv1.CephNFS) error {
+	radosPool := nfs.Spec.RADOS.Pool
+	radosNs := nfs.Spec.RADOS.Namespace
+	radosInfoStr := fmt.Sprintf("rados://%q/%q/", radosPool, radosNs)
+
+	// remove config block that includes the kerberos config object from the ganesha config object
+	ganeshaConfigObjName := getGaneshaConfigObject(nfs)
+	krbIncludeBlock := ganeshaConfigIncludeKrbBlock(nfs, kerberosRadosObjectName)
+	err := atomicRemoveFromConfigObject(context, clusterInfo, radosPool, radosNs, ganeshaConfigObjName, krbIncludeBlock)
+	if err != nil {
+		return errors.Wrap(err, "failed to update the ganesha config object to remove the kerberos object config")
+	}
+
+	// remove the kerberos config rados object
+	cmd := cephclient.NewRadosCommand(context, clusterInfo,
+		[]string{"--pool", radosPool, "--namespace", radosNs, "rm", kerberosRadosObjectName})
+	_, err = cmd.RunWithTimeout(exec.CephCommandsTimeout)
+	if err != nil {
+		return errors.Wrapf(err, "failed to remove the ganesha kerberos config object %s/%s",
+			radosInfoStr, kerberosRadosObjectName)
+	}
+
+	return nil
+}
+
+// prepend a config block to the rados object in an atomic function
+// uses rados locks to ensure the changes we make here aren't going to race with changes that might
+// be made by ceph mgr, ceph itself, or a user manually
+func atomicPrependToConfigObject(
+	context *clusterd.Context, clusterInfo *cephclient.ClusterInfo,
+	radosPool, radosNamespace, objectName, configBlock string,
+) error {
+	tmpFilePattern := fmt.Sprintf("%s_%s_%s_prepend", radosPool, radosNamespace, objectName)
+	objInfoString := fmt.Sprintf("rados://%s/%s/%s", radosPool, radosNamespace, objectName)
+
+	// read object into temp file
+	tempFile, err := ioutil.TempFile("/tmp", tmpFilePattern)
```

Comment:
If you leave this blank, the `osd.TempDir` should be used according to the docs
```suggestion
	tempFile, err := ioutil.TempFile("", tmpFilePattern)
```

Reply:
Perfect! I didn't realize that.